### PR TITLE
felinids are now a quirk instead of a race

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
Felinids are now a 0 point quirk instead of a race

## Why It's Good For The Game
felinids are basically humans, but with some snowflake shit tacked on. this moves some of that to being human, and removes a good deal of other snowflake code (see changelog)

## Changelog
:cl:
add: Body Modder quirk. Gives cat ears, tail, tastes, and a few minor special interactions
del: Felinids
del: mass purrbation. normal purrbation still exits
del: felinid species change sources, including the virus (which i will readd as a true viro symptom  with a preset for admins soon-ish)

tweak: anyone can wag tails now
/:cl:

### CODE REVIEW

**Comment by ivanmixo** [comment]
Does this keep the saymod?

**Comment by Effemu** [comment]
wait does that mean asimov works on felinids now??

**Comment by TeomanTheGreat** [comment]
Does this make them human derived mutants like hulks? If not it should.

**Comment by PowerfulBacon** [comment]
This feels worse than having them as their own species, even if the species is very similar to human since at least all of the species are in 1 place rather than 1 species that is now a snowflake species.

How are felinids different to the other species that warrant them not being a species?

How does this improve code quality even remotely? (Wouldn't having all species in 1 place be better for code quality than just randomly throwing them out all over the place)

The remove 'snowflake' code is basically all the felinid changes.

I get that felinids don't have a lot of changes to human, but it still works best as a species subtype of human since its what they are.

So yea, my big question is but why?

**Comment by PowerfulBacon** [comment]
> 
> 
> Does this keep the saymod?

no because they are humans instead of their own species and saymod is tied to species.

File: code/datums/traits/neutral.dm [hunk 1]
```diff
@@ -106,3 +106,46 @@
...
+	lose_text = "<span class='notice'>You've outgrown your catperson phase</span>"
+
+/datum/quirk/felinid/add()
+	var/mob/living/carbon/human/H = quirk_holder
+	if(H)
+		var/datum/species/species = H.dna.species
+		species.liked_food |= DAIRY
+		species.liked_food |= MEAT
+		species.disliked_food |= SUGAR
+		species.disliked_food |= VEGETABLES
```
Comment (PowerfulBacon) [inline code/datums/traits/neutral.dm:125]: This alone is more snowflake code than the entire species itself and not how species should be used.

**Comment by PowerfulBacon** [comment]
Does this allow IPC felinids?

**Comment by PowerfulBacon** [comment]
The only part of this PR that makes sense and improves the quality of the code is the tail wagging changes.

**Comment by ItsTisme** [comment]
> Does this allow IPC felinids?

cursed

File: code/datums/traits/neutral.dm (hunk 1 above)
Comment (zeskorion) [reply code/datums/traits/neutral.dm:125]: Just works like other taste change prs

**Comment by zeskorion** [comment]
> This feels worse than having them as their own species, even if the species is very similar to human since at least all of the species are in 1 place rather than 1 species that is now a snowflake species.
> 
> How are felinids different to the other species that warrant them not being a species?
> 
> How does this improve code quality even remotely? (Wouldn't having all species in 1 place be better for code quality than just randomly throwing them out all over the place)
> 
> The remove 'snowflake' code is basically all the felinid changes.
> 
> I get that felinids don't have a lot of changes to human, but it still works best as a species subtype of human since its what they are.
> 
> So yea, my big question is but why?

This idea came from talking with franc. Felinids are essentially just humans with cat ears and tail,  and dont need to exist as their own species. So why do they? I only kept the taste bits and the traits because people would be angy if i didnt, but im fine with making it *just* ears and tail

I plan to make a pr... soon(?) That will even further abolish species and make it little more than a template that applies some traits and stuff to you anyways. Fuck species.

**Comment by zeskorion** [comment]
> Does this allow IPC felinids?

Yes. Intentionally so.

**Comment by ivanmixo** [comment]
Keep the say verb please. Tie it into a special tounge, maybe?

**Comment by zeskorion** [comment]
> Keep the say verb please. Tie it into a special tounge, maybe?

I could.

**Comment by PowerfulBacon** [comment]
> This idea came from talking with franc.

That doesn't make it a good idea.

> Felinids are essentially just humans with cat ears and tail, and dont need to exist as their own species. So why do they?

Felinids are reskinned humans with a few unique traits. There is a system in the game for adding in variations to humans which have different skins and special traits and its the species system.

This takes relatively ok species code and makes it into a snowflakey mess of a quirk that accesses and modifies species anyway. If you are modifying a species to act like a different species, then why is that better than having it as the original species.

**Comment by zeskorion** [comment]
> > This idea came from talking with franc.
> 
> That doesn't make it a good idea.
> 
> > Felinids are essentially just humans with cat ears and tail, and dont need to exist as their own species. So why do they?
> 
> Felinids are reskinned humans with a few unique traits. There is a system in the game for adding in variations to humans which have different skins and special traits and its the species system.
> 
> This takes relatively ok species code and makes it into a snowflakey mess of a quirk that accesses and modifies species anyway. If you are modifying a species to act like a different species, then why is that better than having it as the original species.

There's actually no reason for me to change their taste vars and i can just as easily not do so.  Felinids dont need to be a special thing. Let people play with cat ears and tail, whatever, it doesnt need to be an entire species. Changing a minor species var doesnt change shit 

The quirk isnt a snowflakey mess. It's simple. It changes organs and tastes (the tastes in the same way other quirks do, as well). I can point to any number of other quirks that are more of a mess- pacifist and paraplegic come to mind

**Comment by zeskorion** [comment]
I'm fine with removing or adding things to the quirk, however.

**Comment by TeomanTheGreat** [comment]
> > Does this allow IPC felinids?
> 
> Yes. Intentionally so.

Why. Seriously why. Robots shouldn't have cat ears and tails.

**Comment by zeskorion** [comment]
> > > Does this allow IPC felinids?
> > 
> > 
> > Yes. Intentionally so.
> 
> Why. Seriously why. Robots shouldn't have cat ears and tails.

Why not

**Comment by PowerfulBacon** [comment]
> The quirk isnt a snowflakey mess. It's simple. It changes organs and tastes (the tastes in the same way other quirks do, as well). I can point to any number of other quirks that are more of a mess- pacifist and paraplegic come to mind

Yea and species are simpler and made exactly for what felinids have changed from humans.

**Comment by zeskorion** [comment]
> Yea and species are simpler and made exactly for what felinids have changed from humans.

so... two organs and, recently, a saymod. and tastes, which dont matter much. this is all, other than tastes, able to be handled at the organ level- there are no icon differences, there are no stat differences. species are unecessary here

**Comment by TeomanTheGreat** [comment]
> > Yea and species are simpler and made exactly for what felinids have changed from humans.
> 
> so... two organs and, recently, a saymod. and tastes, which dont matter much. this is all, other than tastes, able to be handled at the organ level- there are no icon differences, there are no stat differences. species are unecessary here

They get stunned by laserpointers. Also i think flashbangs stun them longer or something but idk.

**Comment by Sinestia** [comment]
> > > Does this allow IPC felinids?
> > 
> > 
> > Yes. Intentionally so.
> 
> Why. Seriously why. Robots shouldn't have cat ears and tails.

glue


 Honestly bay style of just messing with limbs would be the best imo instead of using the quirk system

**Comment by Effemu** [comment]
>IPC felinids

we're going on the path of vorestation😐

**Comment by zeskorion** [comment]
> > > Yea and species are simpler and made exactly for what felinids have changed from humans.
> > 
> > 
> > so... two organs and, recently, a saymod. and tastes, which dont matter much. this is all, other than tastes, able to be handled at the organ level- there are no icon differences, there are no stat differences. species are unecessary here
> 
> They get stunned by laserpointers. Also i think flashbangs stun them longer or something but idk.

the first is kept via a trait, and the second is attached to the ears



> > > > Does this allow IPC felinids?
> > > 
> > > 
> > > Yes. Intentionally so.
> > 
> > 
> > Why. Seriously why. Robots shouldn't have cat ears and tails.
> 
> glue
> 
> Honestly bay style of just messing with limbs would be the best imo instead of using the quirk system

not feasible

**Comment by francinum** [comment]
@PowerfulBacon That is not what species were made for.

Species are for, well, *species*. They were not split off from humans for code reasons, TG split them off for policy reasons. Questionable policy reasons at that. Cats are also pre-trait system, frankly I agree that at the moment that some of this code isn't all that great, but that's mostly due to poorly located variables, tastes should, if anything, be on the stomach or mind holder.

As of right now, they exist in the strange limbo-state of being human, but not, but yes, but no.

That's really not a good place for anything to be. The species system wasn't really designed to handle subtypes all that well.

**Comment by zeskorion** [comment]
> @PowerfulBacon That is not what species were made for.
> 
> Species are for, well, _species_. They were not split off from humans for code reasons, TG split them off for policy reasons. Questionable policy reasons at that. Cats are also pre-trait system, frankly I agree that at the moment that some of this code isn't all that great, but that's mostly due to poorly located variables, tastes should, if anything, be on the stomach or mind holder.
> 
> As of right now, they exist in the strange limbo-state of being human, but not, but yes, but no.
> 
> That's really not a good place for anything to be. The species system wasn't really designed to handle subtypes all that well.

I think i may try to remove species altogether as they are if i get around to my species refactor tbh. Make "species" a template that gets applied to a mob once, and then the rest of the vars stay with that mob

**Comment by PowerfulBacon** [comment]
> @PowerfulBacon That is not what species were made for.
> 
> Species are for, well, _species_. They were not split off from humans for code reasons, TG split them off for policy reasons. Questionable policy reasons at that. Cats are also pre-trait system, frankly I agree that at the moment that some of this code isn't all that great, but that's mostly due to poorly located variables, tastes should, if anything, be on the stomach or mind holder.
> 
> As of right now, they exist in the strange limbo-state of being human, but not, but yes, but no.
> 
> That's really not a good place for anything to be. The species system wasn't really designed to handle subtypes all that well.

But in this situation species are definitely better than the trait system since it has the added benefit of well, making all species in the species place rather than 1 species which is for some reason in the quirk system.

**Comment by OceanFish1** [comment]
monkey sees 
>del: Felinids

neuron activation

**Comment by Sinestia** [comment]
> > @PowerfulBacon That is not what species were made for.
> > Species are for, well, _species_. They were not split off from humans for code reasons, TG split them off for policy reasons. Questionable policy reasons at that. Cats are also pre-trait system, frankly I agree that at the moment that some of this code isn't all that great, but that's mostly due to poorly located variables, tastes should, if anything, be on the stomach or mind holder.
> > As of right now, they exist in the strange limbo-state of being human, but not, but yes, but no.
> > That's really not a good place for anything to be. The species system wasn't really designed to handle subtypes all that well.
> 
> But in this situation species are definitely better than the trait system since it has the added benefit of well, making all species in the species place rather than 1 species which is for some reason in the quirk system.

the point is that they aren't a species with this 

they're traits that can be applied agnostically, moth cats, etc..

File: code/datums/traits/neutral.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/datums/traits/neutral.dm:125]: The other taste change quirks have no other way to do it. This one has a much much better way to do it; by having them as a species.

**Comment by PowerfulBacon** [comment]
I get the point for it, its not like I'm not understanding. Its just stupid.

**Review by francinum — APPROVED** [review]

**Comment by utilbird** [comment]
No, bad.

Basically everything bacon said is right. I don't really care about the mechanical implications because they're small at best but it's a very bad solution in a development/code sense.

They still have race-like features but unlike every other race, it's handled in traits. If you want to do this you should do it for all of them (which would be even worse than the system we currently have).
Right now this is an objective downgrade in code quality.

> felinids are basically humans, but with some snowflake shit tacked on.

Aaand they still are. Except now it's a bitch and a half to maintain because any unique interactions has to be handled differently to every other race-like interactions.

if you want to get technical, all races are snowflake code and most people are fine with that. Moth content is mostly moth memes, lizards are mostly just lizard memes. 
Speciescode is a code and mechanical improvement over traits and for now I'd rather keep it that way.

**Comment by zeskorion** [comment]
> Speciescode is a code and mechanical improvement over traits and for now I'd rather keep it that way.

Then you sure wont like what i plan to do next 😼

**Comment by AtlasRules** [comment]
just re-open the monkey PR and try that again please 

revert to old zams.... pls.....

**Comment by zeskorion** [comment]
> just re-open the monkey PR and try that again please
> 
> revert to old zams.... pls.....

Old zams is dead and i killed him.

**Pull request closed without merging**